graphite/graphite.go

```diff
@@ -42,9 +42,20 @@ func (r *Request) Query(host string) (Response, error) {
 	if r.End != nil {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
+	u1, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+	if u1.Host == "" && u1.Scheme != "" {
```

Comment:
I want to be much more stringent for this first case. Let's make it

```
if u1.Host == "" && u1.Scheme == "" && u1.Path != "" {
	u1.Host = host
	u1.Scheme = "http"
} else if u1.Scheme == "" || u1.Host == "" || u1.Path != "" {
  return nil, fmt.Errorf("bad URL format: %v", host)
}
```

This ensures that if you just have a hostname:port, it'll use http, and otherwise it requires a valid scheme://host URL with no /path at the end.